integration/getmore_compat_test.go

```diff
@@ -0,0 +1,257 @@
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers: []shareddata.Provider{shareddata.Int32BigAmounts},
+	})
+
+	// We expect to have only one collection as the result of setup.
+	require.Len(t, s.TargetCollections, 1)
+	require.Len(t, s.CompatCollections, 1)
+
+	targetCollection := s.TargetCollections[0]
+	compatCollection := s.CompatCollections[0]
+
+	ctx := s.Ctx
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Helper()
+			t.Run(targetCollection.Name(), func(t *testing.T) {
+				t.Helper()
+
+				var batchSize int32
+				if tc.batchSize != 0 {
+					batchSize = int32(tc.batchSize)
+				}
+				opts := options.Find().SetBatchSize(batchSize)
+
+				var limit int64
+				if tc.limit != 0 {
+					limit = int64(tc.limit)
+				}
+				opts = opts.SetLimit(limit)
+
+				targetResult, targetErr := targetCollection.Find(ctx, bson.D{}, opts)
+				compatResult, compatErr := compatCollection.Find(ctx, bson.D{}, opts)
+
+				if targetErr != nil {
+					t.Logf("Target error: %v", targetErr)
+					AssertMatchesCommandError(t, compatErr, targetErr)
+
+					return
+				}
+
+				require.NoError(t, targetErr)
+				require.NoError(t, compatErr, "compat error; target returned no error")
+
+				// Retrieve all documents from the cursor.
+				// Driver will call getMore until the cursor is exhausted.
+				var targetRes, compatRes []bson.D
+				require.NoError(t, targetResult.All(ctx, &targetRes))
+				require.NoError(t, compatResult.All(ctx, &compatRes))
+
+				assert.Equal(t, len(compatRes), len(targetRes), "result length mismatch")
+			})
+		})
+	}
+}
+
+func TestGetMoreCompat(t *testing.T) {
+	setup.SkipForTigrisWithReason(t, "Iterators are not implemented yet")
+
+	t.Parallel()
+
+	testCases := map[string]queryGetMoreCompatTestCase{
+		"batchSize": {
+			batchSize: 200,
+		},
+		"getMoreWithLimitLessThanBatch": {
+			batchSize: 200,
+			limit:     100,
+		},
+		"getMoreWithLimitGreaterThanBatch": {
+			batchSize: 200,
+			limit:     300,
+		},
+	}
+
+	testGetMoreCompat(t, testCases)
+}
+
+type queryGetMoreErrorsCompatTestCase struct {
+	id         any    // defaults to id fetched from running find command
+	altMessage string // expects different error altMessage on target
+	command    bson.D // defaults to empty
+	err        bool   // expects error if it is set to true
+}
+
+func testGetMoreCompatErrors(t *testing.T, testCases map[string]queryGetMoreErrorsCompatTestCase) {
+	t.Helper()
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers: []shareddata.Provider{shareddata.Int32BigAmounts},
+	})
+
+	// We expect to have only one collection as the result of setup.
+	require.Len(t, s.TargetCollections, 1)
+	require.Len(t, s.CompatCollections, 1)
+
+	targetCollection := s.TargetCollections[0]
+	compatCollection := s.CompatCollections[0]
+
+	ctx := s.Ctx
+
+	for name, tc := range testCases {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
The unexpected errors don't seem to be related to sessions.

Maybe it's a goroutine processing data in parallel because removing the above call to `t.Parallel()` seems to avoid the issue, not sure.